carsus/model/meta/types.py

```diff
@@ -1,52 +1,79 @@
 """Types and SQL constructs specific to carsus"""
 
 from sqlalchemy.ext.hybrid import hybrid_method
-from decimal import Decimal
-from astropy import units as u
-from sqlalchemy import literal
+from sqlalchemy.sql.expression import ClauseElement
+from sqlalchemy.orm.attributes import InstrumentedAttribute
+from astropy.units import Quantity, Unit, dimensionless_unscaled
+import numpy as np
 
-class Quantity(object):
 
-    def __init__(self, value, unit):
-        self.value = value
-        self.unit = unit
+# class Quantity(object):
+#
+#     def __init__(self, value, unit):
+#         self.value = value
+#         self.unit = unit
+#
+#     def __add__(self, other):
+#         return Quantity(
+#                 self.value + other.convert_to(self.unit).value,
+#                 self.unit
+#             )
+#
+#     def __sub__(self, other):
+#         return Quantity(
+#                 self.value - other.convert_to(self.unit).value,
+#                 self.unit
+#             )
+#
+#     def __lt__(self, other):
+#         return self.value < other.convert_to(self.unit).value
+#
+#     def __gt__(self, other):
+#         return self.value > other.convert_to(self.unit).value
+#
+#     def __eq__(self, other):
+#         return self.value == other.convert_to(self.unit).value
+#
+#     @hybrid_method
+#     def convert_to(self, other_unit):
+#         return Quantity(
+#             self.value * self.unit.to(other_unit),
+#             other_unit
+#         )
+#
+#     def __str__(self):
+#         return "%2.4f %s" % (self.value, self.unit)
 
-    def __add__(self, other):
-        return Quantity(
-                self.value + other.convert_to(self.unit).value,
-                self.unit
-            )
 
-    def __sub__(self, other):
-        return Quantity(
-                self.value - other.convert_to(self.unit).value,
-                self.unit
-            )
+class DBQuantity(Quantity):
+    def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
+                subok=False, ndmin=0):
 
-    def __lt__(self, other):
-        return self.value < other.convert_to(self.unit).value
+        if (isinstance(value, InstrumentedAttribute) or
+                isinstance(value, ClauseElement)):
+            if unit is None:
+                unit = dimensionless_unscaled
+            else:
+                unit = Unit(unit)
+
+            value = np.array(value, dtype=dtype, copy=copy, order=order,
+                             subok=False, ndmin=ndmin)
+            value = value.view(cls)
+            value._unit = unit
```

Comment:
I had to override this because sqlalchemy raises a strange error otherwise
```
foo_session = <sqlalchemy.orm.session.Session object at 0x7fa50daf7910>

    def test_atomic_quantity_compare_with_zero(foo_session):
>       res = foo_session.query(AtomicWeight).filter(AtomicWeight.quantity > 0).count()

carsus/model/tests/test_atomic.py:120: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.sql.elements.BinaryExpression object at 0x7fa50daf78d0>

    def __bool__(self):
        if self.operator in (operator.eq, operator.ne):
            return self.operator(hash(self._orig[0]), hash(self._orig[1]))
        else:
>           raise TypeError("Boolean value of this clause is not defined")
E           TypeError: Boolean value of this clause is not defined

../../.virtualenvs/carsus/local/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:2746: TypeError
```
I don't understand yet why the error is raised. It works fine with arithmetic operations, but not with comparisons.